azure-quantum/azure/quantum/job/job.py

```diff
@@ -56,18 +56,30 @@ def has_completed(self) -> bool:
             or self.details.status == "Cancelled"
         )
 
-    def wait_until_completed(self, max_poll_wait_secs=30) -> None:
+    def wait_until_completed(self, max_poll_wait_secs=30, timeout_secs=None) -> None:
```

Comment:
I'm not sure, do we want to give some default timeout, say 5 mins?

Reply:
That would be a breaking change, let's double check with the team if this is a desired change for QIO jobs.

Reply:
Set default to 180s